src/cmap/connection_pool_events.ts

```diff
@@ -202,6 +229,7 @@ export class ConnectionCheckOutFailedEvent extends ConnectionPoolMonitoringEvent
     error?: MongoError
   ) {
     super(pool);
+    this.durationMS = Date.now() - (pool.checkOutTime ?? 0);
```

Comment:
Why is the checkout failed event's durationMS based on the pool.checkoutTime, but the succeeded event is based on the connectionCreatedTime?  Shouldn't they both have the same starting time?

Reply:
According to the [spec](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.md#events), they have different starting times. I've included a clarifying comment in the declaration of the `durationMS` variable. 


For `ConnectionCheckedOutEvent` and `ConnectionCheckoutFailedEvent` the spec says:
```
* More specifically, the time elapsed between
* emitting a `ConnectionCheckOutStartedEvent`
* and emitting this event as part of the same checking out.
``` 

For `ConnectionReadyEvent` the spec says:
```
* The time it took to establish the connection.
* In accordance with the definition of establishment of a connection
* specified by `ConnectionPoolOptions.maxConnecting`,
* it is the time elapsed between emitting a `ConnectionCreatedEvent`
* and emitting this event as part of the same checking out
```